tests/phpunit/includes/bootstrap.php

```diff
@@ -44,15 +44,112 @@
 
 $phpunit_version = tests_get_phpunit_version();
 
-if ( version_compare( $phpunit_version, '5.4', '<' ) || version_compare( $phpunit_version, '8.0', '>=' ) ) {
+if ( version_compare( $phpunit_version, '5.7.21', '<' ) || version_compare( $phpunit_version, '8.0', '>=' ) ) {
 	printf(
-		"Error: Looks like you're using PHPUnit %s. WordPress requires at least PHPUnit 5.4 and is currently only compatible with PHPUnit up to 7.x.\n",
+		"Error: Looks like you're using PHPUnit %s. WordPress requires at least PHPUnit 5.7.21 and is currently only compatible with PHPUnit up to 7.x.\n",
```

Comment:
Curious for the reasoning behind the patch version requirement for PHPUnit here. Will earlier versions of 5.7 not work? Is it because of the forward compatibility for `PHPUnit\Framework\(AssertionFailedError|Test|TestSuite)`?

Reply:
Correct. Earlier versions won't work. PHPUnit added some forward-compatibility namespaced classes to the later patch versions in PHPUnit 4.8.x and 5.7.x and there are differences in which namespaced classes are available between the patch versions.
PHPUnit 5.7.21 is the first 5.x version which has enough of the namespaced classes available for the complete set of polyfills to work.

Reply:
Generally speaking, with a PHPUnit version as old as that, I would expect people to use the last version anyway, which is 5.7.27.